File: Day3/53-Maximum-Subarray/Method3.cpp
class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        int bestSumTillNow = nums[0], bestSumSoFar = nums[0], n = nums.size();

        for (int i = 1; i < n; i++)
        {
            bestSumTillNow = max(nums[i], nums[i] + bestSumTillNow);
            bestSumSoFar = max(bestSumTillNow, bestSumSoFar);
        }

        return bestSumSoFar;
    }
};

File: Day3/53-Maximum-Subarray/Method3.java
class Solution {
    public int maxSubArray(int[] nums) {
        int bestSumTillNow = nums[0], bestSumSoFar = nums[0], n = nums.length;

        for (int i = 1; i < n; i++) {
            bestSumTillNow = Math.max(nums[i], nums[i] + bestSumTillNow);
            bestSumSoFar = Math.max(bestSumTillNow, bestSumSoFar);
        }

        return bestSumSoFar;
    }
}